normalizer/normalizer.go

```diff
@@ -32,28 +32,34 @@ func (n normalizer) Normalize(msg mainflux.RawMessage) (NormalizedData, error) {
 	msgs := make([]mainflux.Message, len(normalized.Records))
 	for k, v := range normalized.Records {
 		m := mainflux.Message{
-			Channel:     msg.Channel,
-			Publisher:   msg.Publisher,
-			Protocol:    msg.Protocol,
-			Name:        v.Name,
-			Unit:        v.Unit,
-			StringValue: v.StringValue,
-			DataValue:   v.DataValue,
-			Time:        v.Time,
-			UpdateTime:  v.UpdateTime,
-			Link:        v.Link,
+			Channel:    msg.Channel,
+			Publisher:  msg.Publisher,
+			Protocol:   msg.Protocol,
+			Name:       v.Name,
+			Unit:       v.Unit,
+			Time:       v.Time,
+			UpdateTime: v.UpdateTime,
+			Link:       v.Link,
 		}
 
 		if v.Value != nil {
-			m.Value = *v.Value
+			m.Value = &mainflux.Message_FloatValue{*v.Value}
 		}
 
 		if v.BoolValue != nil {
-			m.BoolValue = *v.BoolValue
+			m.Value = &mainflux.Message_BoolValue{*v.BoolValue}
+		}
+
+		if v.DataValue != "" {
+			m.Value = &mainflux.Message_DataValue{v.DataValue}
+		}
+
+		if v.StringValue != "" {
+			m.Value = &mainflux.Message_StringValue{v.StringValue}
 		}
```

Comment:
You can use switch case for all of these (you can have only one of these per message, right?).